pkg/cloudprovider/suite_test.go

```diff
@@ -454,7 +456,7 @@ var _ = Describe("CloudProvider", func() {
 			ExpectScheduled(ctx, env.Client, pod)
 			Expect(awsEnv.EC2API.CreateFleetBehavior.CalledWithInput.Len()).To(Equal(1))
 			input := awsEnv.EC2API.CreateFleetBehavior.CalledWithInput.Pop()
-			Expect(input.LaunchTemplateConfigs).To(HaveLen(1))
+			Expect(len(input.LaunchTemplateConfigs)).To(BeNumerically(">=", 1))
```

Comment:
It seems like a lot of our testing logic relies on a single launch template and validating that that first launch template fulfills a given condition. Should we change our logic now to ensure that _all_ launch templates now fulfill this condition?

Reply:
The logic should be updated everywhere to check all LTs in this PR. I've started writing some higher-level expectations  for LTs that we could use to trim the code down a bit though.

Reply:
Is that meant to be a follow-up item to this PR?

Reply:
yep, updated